flow360/component/simulation/unit_system.py

```diff
@@ -219,9 +219,10 @@ def _unit_inference_validator(value, dim_name, is_array=False):
         unit = unit_system_manager.current[dim_name]
         if is_array:
             if all(isinstance(item, Number) for item in value):
-                return value * unit
+                float64_tuple = tuple(np.float64(item) for item in value)
+                return float64_tuple * unit
         if isinstance(value, Number):
-            return value * unit
+            return np.float64(value) * unit
```

Comment:
Why are these float64 needed when we are calling _enforce_float64 anyways?

Reply:
Because we cannot guarantee that `_enforce_flaot64` is always called after these functions. What if this function is used as standalone somewhere?